### PR TITLE
eth, les: reject light client connection is server is not synced

### PR DESCRIPTION
**Pull request by rjl493456442**
This PR introduces protection for les server which will reject all light client connection
if the node is not synced.

Since light client usually tries to retrieve and verify the data based on the latest block hash. 
If the server is not synced, it will only increase the pressure of server and server can't return 
any meaningful response.

### CODE REVIEW

File: les/handler.go [hunk 1]
```diff
@@ -87,12 +87,16 @@ type txPool interface {
 }
 
 type ProtocolManager struct {
-	lightSync    bool
+	// Configs
+	chainConfig *params.ChainConfig
+	iConfig     *light.IndexerConfig
+
+	light     bool   // The indicator whether the node is light client
```
Comment (karalabe) [inline les/handler.go]: Let's call this `client` instead. Since we're in `les`, both server and client are `light`.

File: eth/backend.go [hunk 2]
```diff
@@ -480,6 +480,7 @@ func (s *Ethereum) IsListening() bool                  { return true } // Always
 func (s *Ethereum) EthVersion() int                    { return int(s.protocolManager.SubProtocols[0].Version) }
 func (s *Ethereum) NetVersion() uint64                 { return s.networkID }
 func (s *Ethereum) Downloader() *downloader.Downloader { return s.protocolManager.downloader }
+func (s *Ethereum) IsSynced() bool                     { return atomic.LoadUint32(&s.protocolManager.acceptTxs) == 1 }
```
Comment (karalabe) [inline eth/backend.go]: Please rename it to `Synced()`. Go doesn't generally have the `Is` `Get` prefixes.

File: les/handler.go [hunk 3]
```diff
@@ -268,10 +271,13 @@ func (pm *ProtocolManager) newPeer(pv int, nv uint64, p *p2p.Peer, rw p2p.MsgRea
 func (pm *ProtocolManager) handle(p *peer) error {
 	// Ignore maxPeers if this is a trusted peer
 	// In server mode we try to check into the client pool after handshake
-	if pm.lightSync && pm.peers.Len() >= pm.maxPeers && !p.Peer.Info().Network.Trusted {
+	if pm.light && pm.peers.Len() >= pm.maxPeers && !p.Peer.Info().Network.Trusted {
 		return p2p.DiscTooManyPeers
 	}
-
+	// Reject light clients if server is not synced.
+	if !pm.light && pm.isSynced != nil && !pm.isSynced() {
```
Comment (karalabe) [inline les/handler.go]: Should we test `nil`-ness here? Seems this is a functionality we want to enforce (i.e. not allow it to be nil). For tests I'd rather add a mock method that just returns `true` than make the code weaker.

**Comment by rjl493456442** [comment]
@karalabe Updated.

**Comment by karalabe** [comment]
Bleah, linter fails. @rjl493456442 Please format the file.

**Pull request merged**